File: solutions/erlang/perfect-numbers/1/src/perfect_numbers.erl
-module(perfect_numbers).

-export([classify/1]).


classify(Number) when Number < 1 -> error(nonnatural);
classify(Number) -> 
	Aliquot_sum = aliquot_sum(Number),
	if
		Aliquot_sum < Number -> deficient;
		Aliquot_sum > Number -> abundant;
		true -> perfect
	end.

aliquot_sum(N) -> aliquot_sum(N, 1, 0).

aliquot_sum(N, D, Sum) when 2 * D > N -> Sum;
aliquot_sum(N, D, Sum) when N rem D == 0 -> aliquot_sum(N, D + 1, Sum + D);
aliquot_sum(N, D, Sum) -> aliquot_sum(N, D + 1, Sum).

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError(f"{number} is not a natural number")

    divisors = [n for n in range(1, number // 2 + 1) if number % n == 0]
    aliquot_sum = sum(divisors)
    if aliquot_sum > number:
        return "abundant"
    elif aliquot_sum < number:
        return "deficient"
    else:
        return "perfect"
    